types/color/index.d.ts

```diff
@@ -1,101 +1,135 @@
-// Type definitions for color 1.0
-// Project: https://www.npmjs.com/package/color
-// Definitions by: Karol Janyst <https://github.com/LKay>
+// Type definitions for color 2.0
+// Project: https://github.com/qix-/color#readme
+// Definitions by: Junyoung Clare Jang <https://github.com/Airlun>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
 
-declare namespace Color {
+import convert = require('color-convert');
 
-    type ColorParam = string | { [param: string]: number } | number;
+type ColorParam = Color | string | ArrayLike<number> | number | { [key: string]: any };
 
-    type ColorModel = "rgb" | "hsl" | "hsv" | "hwb" | "hcg" | "cmyk" | "xyz" | "lab" | "hex" | "gray" | "keyword";
+interface Color {
+    toString(): string;
+    toJSON(): Color;
+    string(places?: number): string;
+    percentString(places?: number): string;
+    array(): number[];
+    object(): { alpha?: number } & { [key: string]: number };
+    unitArray(): number[];
+    unitObject(): { r: number, g: number, b: number, alpha?: number };
+    round(places?: number): Color;
+    alpha(): number;
+    alpha(val: number): Color;
+    red(): number;
+    red(val: number): Color;
+    green(): number;
+    green(val: number): Color;
+    blue(): number;
+    blue(val: number): Color;
+    hue(): number;
+    hue(val: number): Color;
+    saturationl(): number;
+    saturationl(val: number): Color;
+    lightness(): number;
+    lightness(val: number): Color;
+    saturationv(): number;
+    saturationv(val: number): Color;
+    value(): number;
+    value(val: number): Color;
+    chroma(): number;
+    chroma(val: number): Color;
+    gray(): number;
+    gray(val: number): Color;
+    white(): number;
+    white(val: number): Color;
+    wblack(): number;
+    wblack(val: number): Color;
+    cyan(): number;
+    cyan(val: number): Color;
+    magenta(): number;
+    magenta(val: number): Color;
+    yellow(): number;
+    yellow(val: number): Color;
+    black(): number;
+    black(val: number): Color;
+    x(): number;
+    x(val: number): Color;
+    y(): number;
+    y(val: number): Color;
+    z(): number;
+    z(val: number): Color;
+    l(): number;
+    l(val: number): Color;
+    a(): number;
+    a(val: number): Color;
+    b(): number;
+    b(val: number): Color;
+    keyword(): string;
+    keyword(val: string): Color;
```

Comment:
@Ailrun I am not familiar with the API and since  the PR is open for more than 7 days, I will merge it if you think this is correct parameter type.